### DISCUSSION TITLE
Typerror: Unknow file extension ".css"

### DISCUSSION THREAD

**Discussion by lcoq** [vitest-dev/vitest#5422] [Q&A] [answered]
This is probably related to #2677 and #5283, but I'm not using `Vuetify` nor react spectrum, just a raw React project.
I don't need CSS at all, but one of the dependencies seems to need it : 

> TypeError: Unknown file extension ".css" for /app/node_modules/@ckeditor/ckeditor5-editor-classic/theme/classiceditor.css

I wonder how can I tell `vitest` to just ignore `.css` files ?

I've tried `css: false` in `vite.config.ts`, but it doesn't change anything, probably because the error says it doesn't know "how" to process css files.

I've also tried this config in `test` :

```
    server: {
      deps: {
        inline: [/@ckeditor.*/]
      }
    }
```

**Comment by hi-ogawa** [comment]
> I wonder how can I tell vitest to just ignore .css files ?

The basic idea of the issue is that NodeJs alone cannot process `.css`, so in order to prevent this, Vitest needs to process the dependencies which use `.css`.

`server.deps.inline` is one way to achieve that. If there's still some error, then NodeJs is still seeing `.css` somewhere. Like the case in https://github.com/vitest-dev/vitest/issues/5283#issuecomment-1962781471, there might be a hidden dependency chain which is not covered by `server.deps.inline`, so running `npm ls @ckeditor/ckeditor5-editor-classic` might help to spot such dependencies.

Or if you can setup a reproduction, then that would be probabaly easier for us to help.

**Comment by lcoq** [comment]
Thank you for your answer !
I've tried to put everything in `server.deps.inline`, even the whole `/@ckeditor\/.*/i`, but it doesn't change anything.

I've made a reproduction setup here : https://stackblitz.com/edit/react-unit-test-practice-cpscyk

Relevant code is IMO  : 

App.tsx

```
import { ClassicEditor } from '@ckeditor/ckeditor5-editor-classic';
import { CKEditor } from '@ckeditor/ckeditor5-react';

return (
  <CKEditor editor={ClassicEditor} />
)
```

**Comment by lcoq** [comment, marked as answer]
Ok got it, I also had to add `/ckeditor5\/.*/i` in deps : https://stackblitz.com/edit/react-unit-test-practice-zbqwhw?file=vite.config.ts

So in the end the `vite.config.ts` looks like this : 

```
export default defineConfig({
  /* ... */
  test: {
    /* ... */
    server: {
      deps: {
        inline: [/@ckeditor\/.*/i, /ckeditor5\/.*/i],
      },
    },
  },
});
```
